src/Value.php

```diff
@@ -172,8 +184,10 @@ public function toStream(): Stream
             return Stream::of($this->get());
         }
 
-        if ($this instanceof Traversable) {
-            return Stream::ofAll($this->iterator()->toArray());
-        }
+        $iterator = $this->iterator();
+
+        return Stream::cons($iterator->next(), function () use ($iterator) {
+            return $iterator->hasNext() ? $iterator->next() : Stream::empty();
```

Comment:
this is huge change, now toStream not evaluate Traversable (allow values to by lazy)